src/Akismet.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Akismet;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Http\Message\ResponseInterface;
+
+class Akismet
+{
+    public $isConfigured = true;
+    private $apiUrl;
+    private $flarumVersion;
+    private $extensionVersion;
+
+    private $params = [];
+    public $proTip;
+
+    public function __construct(string $apiKey, string $homeUrl, $flarumVersion, $extensionVersion, $inDebugMode = false)
+    {
+        if (empty($apiKey)) {
+            return $this->isConfigured = false;
+        }
+
+        $this->apiUrl = "https://$apiKey.rest.akismet.com/1.1";
+        $this->setBlog($homeUrl);
+
+        $this->flarumVersion = $flarumVersion;
+        $this->extensionVersion = $extensionVersion;
+
+        if ($inDebugMode) {
+            $this->setTest();
+        }
+    }
+
+    /**
+     * @param  string  $type  eg. submit-spam, submit-spam or submit-ham;
```

Comment:
"submit-spam" is repeated twice here. Also, since we have methods for `submitSpam` and `submitHam` maybe this should be protected/private?